src/cpp/dtrace/utils.h

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: MIT
+// Copyright (C) 2025 Advanced Micro Devices, Inc. All rights reserved.
+
+#ifndef DTRACE_UTILS_H
+#define DTRACE_UTILS_H
+
+// This file contains the declaration of the log level and expressions for dtrace.
+#ifdef CERT_TRACE_CONTROL_H
+#include "trace_control.h"
+#endif
+
+#include <cstdint>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+
+// Log Levels Enum
+enum class dtrace_log_level {
+    dtrace_error = 1,
+    dtrace_warning = 2,
+    dtrace_info = 3
+};
+
+// Logging Prefix
+#define DTRACE_PREFIX "[DTRACE] "
+
+// Error Logging
+#ifndef DTRACE_ERROR
+#define DTRACE_ERROR(rc, fmt, ...) \
```

Comment:
Why do all macros in this file use do/while? 
Have you considered templatizing this and writing is a C++ code?
I am not pushing for this to be changed now, I am just curious why we are using macro for more that just the injection of __func__, and __line__ ?